### PR TITLE
Rails4 cleaned up + protected attributes instead of strong parameters DO NOT MERGE

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
this is work from the rails4 PR, sans strong parameters. Tests are still failing at the moment, but this looks promising. much smaller diff, small, logical commits.

### CODE REVIEW

File: test/lib/tasks/seeds_test.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class SeedsTest < ActiveSupport::TestCase
     DatabaseCleaner.clean_with :truncation
     Setting.stubs(:[]).with(:administrator).returns("root@localhost")
     Setting.stubs(:[]).with(:send_welcome_email).returns(false)
+    Setting.stubs(:[]).with(:entries_per_page)
```
Comment (dLobatog) [inline test/lib/tasks/seeds_test.rb:12]: Why twice?

**Comment by dLobatog** [comment]
@unorthodoxgeek Mind to rebase against my PRs to develop? That'd clean it up a bit more when they get merged - #2754, #2753, #2751, #2747, #2734,

**Comment by unorthodoxgeek** [comment]
I'm on pto for about 10 days, you can take over if you'd like

**Comment by dLobatog** [comment]
edb11ef5eefcefb242a7f59d52cb0bc5380076b7 is really useful, thanks!

File: app/models/config_group_class.rb [hunk 2]
```diff
@@ -9,7 +9,6 @@ class ConfigGroupClass < ActiveRecord::Base
   belongs_to :puppetclass
   belongs_to :config_group, :counter_cache => true
 
-  validates :puppetclass_id, :presence => true
-  validates :config_group_id, :presence => true,
-                              :uniqueness => {:scope => :puppetclass_id}
+  #validates :puppetclass_id, :presence => true
```
Comment (dLobatog) [inline app/models/config_group_class.rb:12]: Why?

File: app/models/fact_value.rb [hunk 3]
```diff
@@ -17,17 +17,18 @@ class FactValue < ActiveRecord::Base
...
   scope :no_timestamp_facts, lambda {
-    includes(:fact_name).where("fact_names.name <> ?",:_timestamp)
+    includes(:fact_name).where("fact_names.name <> ?",:_timestamp).references(:fact_names)
   }
   scope :timestamp_facts, lambda {
-    joins(:fact_name).where("fact_names.name = ?",:_timestamp)
+    joins(:fact_name).where("fact_names.name = ?",:_timestamp).references(:fact_names)
   }
   scope :my_facts, lambda {
     if !User.current.admin? || Organization.expand(Organization.current).present? || Location.expand(Location.current).present?
```
Comment (dLobatog) [inline app/models/fact_value.rb:26]: Wouldn't an user get fact values for every host (regardless of taxonomies) this way?

**Pull request closed without merging**